### DISCUSSION TITLE
Measure KEM operations execution time

### DISCUSSION THREAD

**Discussion by yamus-didine** [open-quantum-safe/liboqs#1522] [Q&A] [answered]
Hello.
I am working on integrating Kyber in IPsec using Strongswan and Liboqs 0.7.2
I am interested in evaluating the execution time of KEM operations (key generation, encapsulation, and decapsulation). For this purpose, I am trying to use `gettimeofday( )` function that calculates the elapsed time between two points of the code.

- Did I choose the correct method to calculate the execution time (the `gettimeofday( )` function) ?
- If yes, I am confused about where to precisely put the `gettimeofday` begin and end time values of type `timeval`. In `liboqs/src/kem.c` or in `liboqs/src/kem/kyber/.../kem.c` ?

**Comment by dstebila** [comment, marked as answer]
liboqs comes with a timing program (speed_kem) and additionally there is the benchmarking page at https://openquantumsafe.org/benchmarking/.
